pkg/container-collection/k8s.go

```diff
@@ -53,6 +57,12 @@ func NewK8sClient(nodeName string) (*K8sClient, error) {
 
 	fieldSelector := fields.OneTermEqualSelector("spec.nodeName", nodeName).String()
 
+	kubeletConfig, err := getCurrentKubeletConfig(clientset, nodeName)
+	if err != nil {
+		return nil, fmt.Errorf("getting /configz %w", err)
+	}
+	socketPath, _ := strings.CutPrefix(kubeletConfig.ContainerRuntimeEndpoint, "unix:///")
```

Comment:
which mounts are required for socketPath discovery

Reply:
Discovering the socket endpoint doesn't require any new mounts as we find it by getting the kubelet config.
Once we find the socket, In theory, it can be anywhere on the host, but it's most likely to under `/run`.

Reply:
/run is already a host volume, so that should be fine.